### DISCUSSION TITLE
configuring using an environmental variable

### DISCUSSION THREAD

**Discussion by PerryShafran-NOAA** [dtcenter/METplus#1619] [Configuration] [answered]
Hi, there,

Something I'm working on doesn't make sense.  I am trying to run point2grid on a GOES 16 satellite file, with an ADP file, but for some reason, the ADP file isn't being included when point2grid is called.

So in my script, I'm setting a file, which confirmed that it exists:

export adpfile=/gpfs/dell2/emc/modeling/noscrub/Ho-Chun.Huang/GOES16_AOD/ADP/20220514/OR_ABI-L2-ADPC-M6_G16_s20221340001174_e20221340003547_c20221340005467.nc

So in my configuration file (which is /gpfs/dell2/emc/verification/save/Perry.Shafran/METplus-4.0.0/parm/use_cases/perry/point2grid_goes_aod_hrrr.conf if you want to take a look-see), I have the following:

POINT2GRID_ADP = {ENV[adpfile]}

Looks ok thus far?  But in the log file, I see the following:

05/17 14:54:14.762 metplus INFO: COMMAND: /gpfs/dell2/emc/verification/noscrub/emc.metplus/met/10.0.0/exec/point2grid /gpfs/dell2/emc/modeling/noscrub/Ho-Chun.Huang/GOES16_AOD/AOD/20220514/OR_ABI-L2-AODC-M6_G16_s20221340001174_e20221340003547_c20221340007031.nc "/gpfs/hps/nco/ops/com/hrrr/prod/hrrr.20220514/conus/hrrr.t00z.wrfprsf00.grib2" /gpfs/dell2/emc/verification/noscrub/Perry.Shafran/GOES16_AOD_REGRID/20220514/OBS_AOD_hrrr_g16_20220514_00_high.nc -field 'name="AOD_Smoke"; level="";' -adp -method UW_MEAN -gaussian_dx 81.271 -gaussian_radius 120 -v 4
05/17 14:54:14.763 metplus (command_runner.py:122) DEBUG: log_name is: point2grid, output sent to: /gpfs/dell2/emc/verification/noscrub/Perry.Shafran/metplus_hrrraod/logs/metplus.log.20220517145414
OUTPUT:
ERROR  :
ERROR  : CommandLine::get_n_args() -> too few arguments to command-line switch "-adp"
ERROR  :

You'll note that there is nothing after the -adp in the point2grid command.  I'm not sure why that is.  It looks like I have the brackets and braces correct (I had an error last time I wrote in to this forum), so I'm not sure why there is nothing after the -adp when I clearly have that variable set in the METplus config file.

Thanks for any assistance you can provide.

Perry

**Comment by georgemccabe** [comment, marked as answer]
Hi @PerryShafran-NOAA, It looks like there is a bug in the Point2Grid wrapper where the value set for POINT2GRID_ADP is not actually added to the command after the -adp argument.  I created GitHub issue #1623 for this bugfix. Thank you for bringing this to our attention.
